testtools/UART_interface/rpi_uart_interface.py

```diff
@@ -0,0 +1,197 @@
+import os
+import sys
+import time
+
+try:
+    from testtools.UART_interface.base_uart_interface import uart_interface
+    import testtools.UART_interface.azure_test_firmware_errors as azure_test_firmware_errors
+    import testtools.UART_interface.serial_settings as serial_settings
+    import testtools.UART_interface.serial_commands_dict as commands_dict
+except:
+    import azure_test_firmware_errors
+    import serial_settings
+    import serial_commands_dict as commands_dict
+    from base_uart_interface import uart_interface
+
+
+# ------- global usecase fcns -------
+def device_setup():
+    pass
+
+def check_sdk_errors(line):
+    if "Error:" in line:
+        azure_test_firmware_errors.SDK_ERRORS += 1
+
+def check_test_failures(line):
+    if " tests ran" in line:
+        result = [int(s) for s in line.split() if s.isdigit()]
+        azure_test_firmware_errors.SDK_ERRORS = result[1]
+        return result
+
+
+def check_firmware_errors(line):
+    if azure_test_firmware_errors.iot_init_failure in line:
+        print("Failed to connect to saved IoT Hub!")
+        azure_test_firmware_errors.SDK_ERRORS += 1
+
+    elif azure_test_firmware_errors.sensor_init_failure in line:
+        print("Failed to init mxchip sensor")
+        azure_test_firmware_errors.SDK_ERRORS += 1
+
+    elif azure_test_firmware_errors.wifi_failure in line:
+        print("Failed to connect to saved WiFi network.")
+        azure_test_firmware_errors.SDK_ERRORS += 1
+
+# ------- interface class -------
+class rpi_uart_interface(uart_interface):
+
+    # If there is a sudden disconnect, program should report line in input script reached, and close files.
+    # method to write to serial line with connection monitoring
+    def serial_write(self, ser, message, file=None):
+
+        # Check that the device is no longer sending bytes
+        if ser.in_waiting:
+            self.serial_read(ser, message, file)
+
+        # Check that the serial connection is open
+        if ser.writable():
+
+            # special handling for sending a file
+            if "sz " in message:
```

Comment:
>"sz " [](start = 15, length = 5)

nit: style - Let us label these magic commands for now. It would be good to have one file maintaining all command names (along with their input)